pkg/dashboard/apiserver/event/event.go

```diff
@@ -97,6 +106,120 @@ func (s *Service) list(c *gin.Context) {
 	c.JSON(http.StatusOK, events)
 }
 
+// @Summary cascadeFetchEventsForWorkflow list all events for Workflow and related WorkflowNode.
+// @Description list all events for Workflow and related WorkflowNode.
+// @Tags events
+// @Produce json
+// @Param namespace query string false "The namespace of the object"
+// @Param object_id query string false "The UID of the Workflow"
+// @Param limit query string false "The max length of events list"
+// @Success 200 {array} core.Event
+// @Failure 500 {object} utils.APIError
+// @Router /events/workflow [get]
+func (s *Service) cascadeFetchEventsForWorkflow(c *gin.Context) {
+	ctx := c.Request.Context()
+	ns := c.Query("namespace")
+	objectId := c.Query("object_id")
+	limit := 0
+	limitString := c.Query("limit")
+	if len(limitString) > 0 {
+		parsedLimit, err := strconv.Atoi(limitString)
+		if err != nil {
+			c.Status(http.StatusBadRequest)
+			_ = c.Error(u.ErrBadRequest.New("parameter limit should be a integer"))
```

Comment:
Should use `utils.SetAPIError`. The group of `c.Status` and `c.Error` has been replaced by it.

Reply:
fixed!